src/AbstractFirestoreRepository.ts

```diff
@@ -292,6 +297,25 @@ export abstract class AbstractFirestoreRepository<T extends IEntity>
     return new QueryBuilder<T>(this).findOne();
   }
 
+  /**
+   * Uses class-validator to validate an entity using decorators set in the collection class
+   * 
+   * @param item class or object representing an entity
+   * @returns {Promise<ValidationError[]>} An array of class-validator errors
+   */
+  async validate(item: T): Promise<ValidationError[]> {
```

Comment:
Wouldn't `validateModel` or `runModelValidations` be better?

Reply:
Resolving this to move to other comment